tests/interchain/provider_test.go

```diff
@@ -0,0 +1,608 @@
+package interchain
+
+import (
+	"cosmos/interchain-security/tests/interchain/chainsuite"
+	"fmt"
+	"testing"
+	"time"
+
+	govv1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+	providertypes "github.com/cosmos/interchain-security/v6/x/ccv/provider/types"
+	"github.com/strangelove-ventures/interchaintest/v8/chain/cosmos"
+	"github.com/strangelove-ventures/interchaintest/v8/testutil"
+
+	"github.com/stretchr/testify/suite"
+)
+
+func TestProviderSuite(t *testing.T) {
+	s := &ProviderSuite{}
+
+	suite.Run(t, s)
+}
+
+////////////////////////////////////////////////////////////
+//				Chain CRUD flow test					  //
+////////////////////////////////////////////////////////////
+
+// Test Creating a chain (MsgCreateConsumer)
+// Confirm that a chain can be created with the minimum params (only consumer metadata)
+// Confirm that a chain can be created with all params
+// Confirm that a chain can be created with initialization parameters that do not contain a spawn time
+// Confirm that if there are no opted-in validators at spawn time, the chain fails to launch and moves back to its Registered phase having reset its spawn time
+func (s *ProviderSuite) TestProviderCreateConsumer() {
+	// Confirm that a chain can be create with the minimum params (metadata)
+	chainName := "minParamAddConsumer"
+	proposalMsg := msgCreateConsumer(chainName, nil, nil)
+	s.Require().NoError(s.Provider.ExecuteProposalMsg(s.GetContext(), proposalMsg, chainsuite.GovModuleAddress, chainName, cosmos.ProposalVoteYes, govv1.StatusPassed, false))
```

Comment:
Hey, can you change occurences of `.ExecuteProposal()` to only occur when dealing with TopN chains?

Only TopN chains are permissioned, that means that normal chains don't require a proposal for adding, updating params etc.

TopN chain requires a proposal for updating the record, removing it etc.

An opt-in chain can become a TopN via a gov proposal (you have that flow covered, I'm just stating it)

Reply:
fixed [fdf4ca9](https://github.com/cosmos/interchain-security/pull/2368/commits/fdf4ca90e2ba42a1a778f1de2182fc324d336f7d)